ls1intum/Artemis
# [Feature] Learning Goals

### Checklist

### Motivation and Context



### Description

Every course has overarching themes/goals that students should acquire while doing the course. For Example: Learn HTML, or Learn CSS. This feature now allows instructors to define such course goals and connect them to corresponding lecture units. The learning goal progress is then calculated for a student from his/her performance in the connected lecture units (currently only exercises)

### Steps for Testing

Basically the test case would be to test the learning goals CRUD (Create, Read, Update,Delete) functionality from the instructor perspective (create learning goals, connect them to lecture units and so on) and check that the corresponding view from the student perspective is also updated

### Test Coverage




Server
- LearningGoalResource 82%
- LearningGoalService 95%

Client

- create-learning-goal 82%
- edit-learning-goal 95%
- learning-goal-form 89%
- learning-goal-detail-modal 91%
- learning-goal-management 86%
- course-learning-goals 96%
- learning-goal-popver 100%
- learning-goal-card 89%
- circular-progress-bar 100%

### Screenshots



**Instructors**
![image](https://user-images.githubusercontent.com/29718932/101913294-1613f400-3bc3-11eb-8595-910c6839c0ea.png)
![image](https://user-images.githubusercontent.com/29718932/101913319-1f04c580-3bc3-11eb-9280-0ccc034fbee7.png)

![image](https://user-images.githubusercontent.com/29718932/101360564-17c97900-389e-11eb-9782-b40e4813c8ee.png)
![image](https://user-images.githubusercontent.com/29718932/101360715-54957000-389e-11eb-9901-82ecd4628d7a.png)
![image](https://user-images.githubusercontent.com/29718932/101360912-945c5780-389e-11eb-9e17-a19f856ae1e1.png)
![image](https://user-images.githubusercontent.com/29718932/101360938-9cb49280-389e-11eb-9a16-69738f23a4fb.png)
![image](https://user-images.githubusercontent.com/29718932/101361096-d7b6c600-389e-11eb-8e52-1b76561c9891.png)
![image](https://user-images.githubusercontent.com/29718932/101361148-e7360f00-389e-11eb-9c79-702cdfdc78f6.png)
**Students**
![image](https://user-images.githubusercontent.com/29718932/101361255-03d24700-389f-11eb-946d-c5cc9f405ab0.png)
![image](https://user-images.githubusercontent.com/29718932/101361283-0f257280-389f-11eb-998b-d89170e910ce.png)
